lib/prx_auth/certificate_cache.ex

```diff
@@ -0,0 +1,47 @@
+defmodule PrxAuth.Certificate.Cache do
```

Comment:
Yeah, it may be worth using a more generic cache like [con_cache](https://hex.pm/packages/con_cache).  There are other places in this app that I'd use it, like caching BigQuery auth tokens.  I'll open a ticket to make sure that happens when I turn this into a hex module.

(But this was still a good learning experience in how GenServer works).